src/pages/workspace/members/WorkspaceOwnerChangeCheckPage.tsx

```diff
@@ -0,0 +1,110 @@
+import type {StackScreenProps} from '@react-navigation/stack';
+import React, {useCallback, useEffect} from 'react';
+import {View} from 'react-native';
+import type {OnyxEntry} from 'react-native-onyx';
+import {withOnyx} from 'react-native-onyx';
+import type {ValueOf} from 'type-fest';
+import Button from '@components/Button';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import ScreenWrapper from '@components/ScreenWrapper';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as WorkspaceSettingsUtils from '@libs/WorkspacesSettingsUtils';
+import Navigation from '@navigation/Navigation';
+import type {SettingsNavigatorParamList} from '@navigation/types';
+import AdminPolicyAccessOrNotFoundWrapper from '@pages/workspace/AdminPolicyAccessOrNotFoundWrapper';
+import PaidPolicyAccessOrNotFoundWrapper from '@pages/workspace/PaidPolicyAccessOrNotFoundWrapper';
+import type {WithPolicyOnyxProps} from '@pages/workspace/withPolicy';
+import withPolicy from '@pages/workspace/withPolicy';
+import * as PolicyActions from '@userActions/Policy';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type SCREENS from '@src/SCREENS';
+import type {PersonalDetailsList} from '@src/types/onyx';
+
+type WorkspaceOwnershipChangeChecksOnyxProps = {
+    /** Personal details of all users */
+    personalDetails: OnyxEntry<PersonalDetailsList>;
+};
+
+type WorkspaceMemberDetailsPageProps = WithPolicyOnyxProps &
+    WorkspaceOwnershipChangeChecksOnyxProps &
+    StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.OWNER_CHANGE_CHECK>;
+
+function WorkspaceOwnerChangeCheckPage({route, personalDetails, policy}: WorkspaceMemberDetailsPageProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+
+    const policyID = route.params.policyID;
+    const accountID = route.params.accountID;
+    const error = route.params.error;
+
+    useEffect(() => {
+        if (!policy) {
```

Comment:
The logic in this useEffect seems to be the same as the one in `WorkspaceMemberDetailsPage`. Can we extract it into a method and call that method in both places?

Reply:
extracted